webcompat/static/js/lib/bugform.js

```diff
@@ -155,19 +155,40 @@ BugForm.prototype.onReceiveMessage = function(event) {
   // Make sure the data is coming from a trusted source.
   // (i.e., our add-on or some other priviledged code sent it)
   if (location.origin === event.origin) {
-    // See https://github.com/webcompat/webcompat.com/issues/1252 to track
-    // the work of only accepting blobs, which should simplify things.
-    if (event.data instanceof Blob) {
-      // convertToDataURI sends the resulting string to the upload
-      // callback.
-      this.convertToDataURI(event.data, this.showUploadPreview.bind(this));
+    if (
+      event.data.hasOwnProperty("screenshot") ||
+      event.data.hasOwnProperty("message")
+    ) {
+      this.handleScreenshot(event.data.screenshot);
```

Comment:
I'm not sure what the actual risk is, but in theory `event.data` could have a missing `screenshot` member, but have a `message` one. In that case, would `this.handleScreenshot(event.data.screenshot);` throw?

(Hard to say for sure, since right now Firefox doesn't send this object in this format currently (but I think you have local patches that do this).

Reply:
edit: maybe the `||` is intended to be `&&`?

Reply:
@ksy36 this is the last question I guess before merging.

Reply:
Ahh thanks, I'll change this

Reply:
> Ahh thanks, I'll change this

cool! Feel free to merge when you're ready.